### PR TITLE
Feature/queries acceptance tests

### PR DESCRIPTION
**Pull request by Carver182**
### Description of the Change

These tests checks if the user can performs queries just in the allowed time interval and with valid signature and public key attached to the query

### Benefits

Allow to reveal query incorrectness from the point of view invalid Signatures and created time 

### Possible Drawbacks 

none

### Usage Examples or Tests *[optional]*

Just run queries_acceptance_tests

### CODE REVIEW

File: test/integration/acceptance/queries_acceptance_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,695 @@
...
+      .sendTx(makeUserWithPerms(
+          {shared_model::interface::permissions::Role::kGetRoles}))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      .sendQuery(query, checkQuery);
+}
+
+/**
+ * @given query with Keypair which contain invalid sign but valid public key
```
Comment (nickaleks) [inline test/integration/acceptance/queries_acceptance_tests.cpp]: contains invalid signature?

**Comment by igor-egorov** [comment]
DCO will be fixed automatically when new commits pushed.

File: test/integration/acceptance/queries_acceptance_tests.cpp [hunk 2]
```diff
@@ -0,0 +1,695 @@
...
+  };
+  std::string invalidPrivateKey;
+  std::string invalidPublicKey;
+
+  const std::string NonExistentUserId = "aaaa@aaaa";
+};
+
+/**
+ * @given query with a non-existent creator_account_id
+ * @when execute any correct query with needed permitions
```
[the remark this answers is not in the document]
Comment (Carver182) [reply test/integration/acceptance/queries_acceptance_tests.cpp]: +

File: test/integration/acceptance/queries_acceptance_tests.cpp (hunk 1 above)
Comment (Carver182) [reply test/integration/acceptance/queries_acceptance_tests.cpp]: +

File: test/integration/acceptance/queries_acceptance_tests.cpp [hunk 3]
```diff
@@ -0,0 +1,695 @@
...
+using namespace integration_framework;
+using namespace shared_model;
+
+class QueriesAcceptanceTests : public AcceptanceFixture {
+ public:
+  QueriesAcceptanceTests()
+      : invalidPrivateKey(kUserKeypair.privateKey().hex()),
+        invalidPublicKey(kUserKeypair.publicKey().hex()) {
+    invalidPrivateKey[0] == '9' || invalidPrivateKey[0] == 'f'
+        ? invalidPrivateKey[0] = --invalidPrivateKey[0]
```
[the remark this answers is not in the document]
Comment (Carver182) [reply test/integration/acceptance/queries_acceptance_tests.cpp]: +

File: test/integration/acceptance/queries_acceptance_tests.cpp [hunk 4]
```diff
@@ -0,0 +1,695 @@
...
+ * @given query with a non-existent creator_account_id
+ * @when execute any correct query with needed permitions
+ * @then the query should not pass stateful validation
+ */
+TEST_F(QueriesAcceptanceTests, NonExistentCreatorId) {
+  auto checkQuery = [](auto &queryResponse) {
+    ASSERT_NO_THROW({
+      boost::apply_visitor(
+          framework::SpecifiedVisitor<
+              shared_model::interface::StatefulFailedErrorResponse>(),
```
[the remark this answers is not in the document]
Comment (Carver182) [reply test/integration/acceptance/queries_acceptance_tests.cpp]: +

File: test/integration/acceptance/queries_acceptance_test.cpp [hunk 5]
```diff
@@ -0,0 +1,487 @@
...
+  template<typename F>
+  auto checkIfStatefulFailed(F& queryResponse)
+  {
+    ASSERT_TRUE(
+        boost::apply_visitor(interface::QueryErrorResponseChecker<
+                                 interface::StatefulFailedErrorResponse>(),
+                             queryResponse.get()))
+                  << "Actual response: " << queryResponse.toString();
+  };
+  —
```
Comment (igor-egorov) [inline test/integration/acceptance/queries_acceptance_test.cpp]: That character causes compilation error.
```
.../queries_acceptance_test.cpp:56:3: error: non-ASCII characters are not allowed outside of literals and identifiers
  —
  ^
1 error generated.
```

File: test/integration/acceptance/queries_acceptance_test.cpp [hunk 6]
```diff
@@ -0,0 +1,487 @@
...
+  void SetUp() {
+    itf.setInitialState(kAdminKeypair)
+        .sendTx(makeUserWithPerms({interface::permissions::Role::kGetRoles}))
+        .skipProposal()
+        .checkBlock([](auto &block) {
+          ASSERT_EQ(boost::size(block->transactions()), 1);
+        });
+  };
+
+  template<typename F>
```
Comment (igor-egorov) [inline test/integration/acceptance/queries_acceptance_test.cpp]: What is the reason for templates usage?

The type of parameter always is `const shared_model::proto::QueryResponse &`

File: test/integration/acceptance/queries_acceptance_test.cpp [hunk 7]
```diff
@@ -0,0 +1,487 @@
...
+  auto checkIfStatefulFailed(F& queryResponse)
```
Comment (igor-egorov) [inline test/integration/acceptance/queries_acceptance_test.cpp]: All three verification functions do not access any class members of the fixture.
Making them static will let you eliminate usage of such proxy lambdas:
```
auto check = [this](auto &queryResponse){checkIfStatefulFailed(queryResponse);};
```

Please make them static and remove proxy lambdas.
The usage would be quite simple:
```
itf.sendQuery(query, checkIfStatefulFailed);
```

File: test/integration/acceptance/queries_acceptance_test.cpp [hunk 8]
```diff
@@ -0,0 +1,466 @@
...
+/**
+ * @given query with a non-existent creator_account_id
+ * @when execute any correct query with kGetRoles permissions
+ * @then the query should not pass stateful validation
+ */
+TEST_F(QueriesAcceptanceTest, NonExistentCreatorId) {
+
+  auto check = [this](auto &queryResponse){checkIfStatefulFailed(queryResponse);};
+
+  auto query = baseQry()
```
Comment (igor-egorov) [inline test/integration/acceptance/queries_acceptance_test.cpp]: Why have you changed TestUnsignedQueryBuilder to baseQry at the only place?

Comment (igor-egorov) [reply test/integration/acceptance/queries_acceptance_test.cpp]: Usage of baseQry and complete will let you reduce the amount of code to be typed. Please check thier implementations.

For example, here query creation will look like as follows:
```
auto query = complete(baseQry(NonExistentUserId).getRoles());
```
In order to use that feature `baseQry(account_id)` you have to merge the latest develop to this branch.

File: test/integration/acceptance/queries_acceptance_test.cpp [hunk 9]
```diff
@@ -0,0 +1,466 @@
...
+/**
+ * @given query with an 1 hour old UNIX time
+ * @when execute any correct query with kGetRoles permissions
+ * @then the query returns list of roles
+ */
+TEST_F(QueriesAcceptanceTest, OneHourOldTime) {
+
+  auto check = [this](auto &queryResponse){checkIfSuccess(queryResponse);};
+
+  auto query = TestUnsignedQueryBuilder()
```
Comment (igor-egorov) [inline test/integration/acceptance/queries_acceptance_test.cpp]: Change the rest cases too.
`complete` can also take custom keypair as the second parameter (in the latest develop).

File: test/integration/acceptance/queries_acceptance_test.cpp [hunk 10]
```diff
@@ -0,0 +1,466 @@
...
+
+  itf.sendQuery(query, check);
+}
+
+/**
+ * @given query with an 1 hour old UNIX time
+ * @when execute any correct query with kGetRoles permissions
+ * @then the query returns list of roles
+ */
+TEST_F(QueriesAcceptanceTest, OneHourOldTime) {
```
Comment (igor-egorov) [inline test/integration/acceptance/queries_acceptance_test.cpp]: The whole test case may look like:
```
TEST_F(QueriesAcceptanceTest, OneHourOldTime) {
  auto query =
      complete(baseQry()
                   .createdTime(iroha::time::now(std::chrono::hours(-1)))
                   .getRoles());

  itf.sendQuery(query, checkIfSuccess);
}
```

File: test/integration/acceptance/queries_acceptance_test.cpp [hunk 11]
```diff
@@ -0,0 +1,466 @@
...
+using namespace shared_model;
+
+class QueriesAcceptanceTest : public AcceptanceFixture {
+ public:
+  QueriesAcceptanceTest()
+      : itf(1),
+        invalidPrivateKey(kUserKeypair.privateKey().hex()),
+        invalidPublicKey(kUserKeypair.publicKey().hex()) {
+    /*
+     * It's deliberately get broken the public key and privite key to simulate a
```
Comment (nickaleks) [inline test/integration/acceptance/queries_acceptance_test.cpp]: More correct sentence: "deliberately break the public and private keys to simulate..."

**Pull request merged**